soc/riscv/riscv-privileged/common/soc_common.h

```diff
@@ -41,6 +41,59 @@
 void soc_interrupt_init(void);
 #endif
 
+#if defined(CONFIG_RISCV_CUSTOM_INTERRUPT_CONTROLLER)
+
+/**
+ * @brief Enable an interrupt line for RISC-V custom interrupt controller.
+ *
+ * This routine enables an interrupt line for RISC-V custom interrupt controller.
+ * Should be implemented by custom interrupt controller.
+ *
+ * @param irq IRQ number to enable
+ */
+void z_soc_irq_enable(uint32_t irq);
+
+/**
+ * @brief Disable an interrupt line for RISC-V custom interrupt controller.
+ *
+ * This routine disables an interrupt line for RISC-V custom interrupt controller.
+ * Should be implemented by custom interrupt controller.
+ *
+ * @param irq IRQ number to disable
+ */
+void z_soc_irq_disable(uint32_t irq);
+
+/**
+ * @brief Check if an interrupt line for RISC-V custom interrupt controller is enabled
+ *
+ * This routine checks if an interrupt line for RISC-V custom interrupt
+ * controller is enabled.
+ * Should be implemented by custom interrupt controller.
+ *
+ * @param irq IRQ number to check
+ *
+ * @return 1 - if IRQ is enabled; 0 - otherwise
+ */
+int z_soc_irq_is_enabled(uint32_t irq);
+
+/**
+ * @brief Set priority of a RISC-V custom interrupt controller interrupt line
+ *
+ * This routine set the priority of a RISC-V custom interrupt controller
+ * interrupt line. Should be implemented by custom interrupt controller.
+ *
+ * @param irq IRQ number for which to set priority
+ * @param priority Priority of IRQ to set to
+ */
+void z_soc_irq_priority_set(uint32_t irq, uint32_t priority);
```

Comment:
did you even try to compile this code? How this possibly could compile correctly when this function is missing the `flags` parameter?

Reply:
My bad, thanks! Missed it

Reply:
please, always at least compile-test your PRs

Reply:
That's a shame! I've built it for qemu_riscv64_smp, but didn't try to enable the config option.